### DISCUSSION TITLE
Support for custom renderers

### DISCUSSION THREAD

**Discussion by jacobc94** [AvaloniaUI/Avalonia.Samples#77] [Q&A] [answered]
Hello,
I am exploring Uno and Avalonia as alternatives to MAUI and I am wondering if Avalonia supports custom renderers, like this one in windows:
`            protected override void OnPaint(PaintEventArgs e) 
`
If you could let me know, that would be great. Thank you!

**Comment by maxkatz6** [comment, marked as answer]
Yes, it's:
```csharp
public override void Render(DrawingContext context)
{
        context.FillRectangle(Brushes.White, Bounds);
}
```

Just inherit Control type and you will be able to make custom draw operations.

**Comment by timunie** [comment]
@jacobc94 See also this ticket: https://github.com/AvaloniaUI/Avalonia.Samples/issues/73 

Imo, Avalonia is very powerful when it comes to customizations. From other feedback I got even more powerful than MAUI.
